dnf-behave-tests/features/steps/common/file.py

```diff
@@ -35,6 +74,17 @@ def create_file_with_contents(filename, contents, encoding="utf-8"):
         outstream.flush()
     assert os.path.exists(filename), "ENSURE: file exists {!r}".format(filename)
 
+def regex_read_file_contents(filename, encoding="utf-8"):
```

Comment:
Rather reuse current read_file_contents. You can extend it to be able to cope with compressed files, but I do not see strong reason for creating this new function.